gobblin-config-management/gobblin-config-core/src/main/java/gobblin/config/store/hdfs/SimpleHDFSConfigStore.java

```diff
@@ -0,0 +1,374 @@
+        try (InputStream includesConfInStream = this.fs.open(includesFileStatus.getPath())) {
+          configKeyPaths.addAll(Lists.newArrayList(
+              Iterables.transform(IOUtils.readLines(includesConfInStream, Charsets.UTF_8), new IncludesToConfigKey())));
+        }
+      }
+    } catch (IOException e) {
+      throw new RuntimeException(String.format("Error while getting imports for configKey: \"%s\"", configKey), e);
+    }
+    return configKeyPaths;
+  }
+
+  /**
+   * Retrieves the {@link Config} for the given {@link ConfigKeyPath} by reading the {@code main.conf} associated with
+   * the dataset specified by the given {@link ConfigKeyPath}. If the {@link Path} described by the {@link ConfigKeyPath}
+   * does not exist then an empty {@link Config} is returned.
+   *
+   * @param  configKey      the config key path whose properties are needed.
+   * @param  version        the configuration version in the configuration store.
+   *
+   * @return a {@link Config} for the given configKey.
+   *
+   * @throws VersionDoesNotExistException if the version specified cannot be found in the {@link ConfigStore}.
+   */
+  @Override
+  public Config getOwnConfig(ConfigKeyPath configKey, String version)
+      throws VersionDoesNotExistException {
+    Preconditions.checkNotNull(configKey, "configKey cannot be null!");
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(version), "version cannot be null or empty!");
+
+    Path datasetDir = getDatasetDirForKey(configKey, version);
+    Path mainConfFile = new Path(datasetDir, "main.conf");
+
+    try {
+      if (!this.fs.exists(mainConfFile)) {
+        return ConfigFactory.empty();
+      }
+
+      FileStatus configFileStatus = this.fs.getFileStatus(mainConfFile);
+      if (!configFileStatus.isDir()) {
+        try (InputStream mainConfInputStream = this.fs.open(configFileStatus.getPath())) {
+          return ConfigFactory.parseReader(new InputStreamReader(mainConfInputStream));
+        }
+      } else {
+        return ConfigFactory.empty();
+      }
+    } catch (IOException e) {
+      throw new RuntimeException(String.format("Error while getting config for configKey: \"%s\"", configKey), e);
+    }
+  }
+
+  /**
+   * Retrieves the dataset dir on HDFS associated with the given {@link ConfigKeyPath} and the given version. This
+   * directory contains the {@code main.conf} and {@code includes.conf} file, as well as any child datasets.
+   */
+  private Path getDatasetDirForKey(ConfigKeyPath configKey, String version) throws VersionDoesNotExistException {
+    System.out.println("getVersionRoot(version) " + getVersionRoot(version));
+    System.out.println("getDatasetFromConfigKey(configKey) " + getDatasetFromConfigKey(configKey));
+    return new Path(getVersionRoot(version), getDatasetFromConfigKey(configKey));
+  }
+
+  /**
+   * Retrieves the name of a dataset from a given {@link ConfigKeyPath}.
+   */
+  private String getDatasetFromConfigKey(ConfigKeyPath configKey) {
```

Comment:
configKey is relative path, not absolute, for example; configKeyPath is /data/databases
but store root will be etl-hdfs://eat1-nertznn01.grid.linkedin.com:9000/user/mitu/HdfsBasedConfigTest/